src/CommunityToolkit.Maui/Behaviors/ProgressBarAnimationBehavior.shared.cs

```diff
@@ -7,29 +7,90 @@ namespace CommunityToolkit.Maui.Behaviors;
 /// /// </summary>
 public class ProgressBarAnimationBehavior : BaseBehavior<ProgressBar>
 {
+	readonly WeakEventManager animationCompletedEventManager = new();
+
+	/// <summary>
+	/// Backing BindableProperty for the <see cref="Progress"/> property.
+	/// </summary>
+	public static readonly BindableProperty ProgressProperty =
+		BindableProperty.CreateAttached(nameof(Progress), typeof(double), typeof(ProgressBarAnimationBehavior), 0.0d, propertyChanged: OnAnimateProgressPropertyChanged);
+
 	/// <summary>
-	/// Backing BindableProperty for the <see cref="AnimateProgress"/> property.
+	/// BindableProperty for the <see cref="Length"/> property
 	/// </summary>
-	public static readonly BindableProperty AnimateProgressProperty =
-		BindableProperty.CreateAttached(nameof(AnimateProgress), typeof(double), typeof(ProgressBar), 0.0d, propertyChanged: OnAnimateProgressPropertyChanged);
+	public static readonly BindableProperty LengthProperty =
+		BindableProperty.CreateAttached(nameof(Length), typeof(uint), typeof(ProgressBarAnimationBehavior), (uint)500);
 
 	/// <summary>
-	/// Animation Progress, 0-1.0
+	/// BindableProperty for the <see cref="Easing"/> property
 	/// </summary>
-	public double AnimateProgress
+	public static readonly BindableProperty EasingProperty =
+		BindableProperty.CreateAttached(nameof(Easing), typeof(Easing), typeof(ProgressBarAnimationBehavior), Easing.Linear);
+
+	/// <summary>
+	/// Event that is triggered when the ProgressBar.ProgressTo() animation completes
+	/// </summary>
+	public event EventHandler AnimationCompleted
 	{
-		get => (double)GetValue(AnimateProgressProperty);
-		set => SetValue(AnimateProgressProperty, value);
+		add => animationCompletedEventManager.AddEventHandler(value);
+		remove => animationCompletedEventManager.RemoveEventHandler(value);
 	}
 
-	static async void OnAnimateProgressPropertyChanged(BindableObject bindable, object oldValue, object newValue)
-		=> await ((ProgressBarAnimationBehavior)bindable).Animate();
+	/// <summary>
+	/// Value of <see cref="ProgressBar.Progress"/>, can have a minimum value of 0 and a maximum value of 1
+	/// </summary>
+	public double Progress
+	{
+		get => (double)GetValue(ProgressProperty);
+		set
+		{
+			if (value < 0 || value > 1)
+			{
+				throw new ArgumentOutOfRangeException(nameof(Progress), $"{nameof(Progress)} can have a minimum value of 0 and a maximum value of 1");
```

Comment:
Should we throw exception? According to XF docs: Progress values less than 0 will be clamped to 0, values greater than 1 will be clamped to 1.

https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/progressbar

Reply:
Good catch! We should definitely follow the .NET MAUI team's lead 💯